SwiftCheck/Arbitrary.swift

```diff
@@ -479,47 +479,55 @@ extension HalfOpenInterval where Bound : protocol<Comparable, Arbitrary> {
 	}
 }
 
-extension ImplicitlyUnwrappedOptional where T : Arbitrary {
-	public static var arbitrary : Gen<ImplicitlyUnwrappedOptional<T>> {
-		return ImplicitlyUnwrappedOptional.init <^> Optional<T>.arbitrary
+extension ImplicitlyUnwrappedOptional where Wrapped : Arbitrary {
+	public static var arbitrary : Gen<ImplicitlyUnwrappedOptional<Wrapped>> {
+		return ImplicitlyUnwrappedOptional.init <^> Optional<Wrapped>.arbitrary
 	}
 	
-	public static func shrink(bl : ImplicitlyUnwrappedOptional<T>) -> [ImplicitlyUnwrappedOptional<T>] {
-		return Optional<T>.shrink(bl).map(ImplicitlyUnwrappedOptional.init)
+	public static func shrink(bl : ImplicitlyUnwrappedOptional<Wrapped>) -> [ImplicitlyUnwrappedOptional<Wrapped>] {
+		return Optional<Wrapped>.shrink(bl).map(ImplicitlyUnwrappedOptional.init)
 	}
 }
 
 extension ImplicitlyUnwrappedOptional : WitnessedArbitrary {
-	public typealias Param = T
+	public typealias Param = Wrapped
 	
-	public static func forAllWitnessed<A : Arbitrary>(wit : A -> T)(pf : (ImplicitlyUnwrappedOptional<T> -> Testable)) -> Property {
+	public static func forAllWitnessed<A : Arbitrary>(wit : A -> Wrapped)(pf : (ImplicitlyUnwrappedOptional<Wrapped> -> Testable)) -> Property {
 		return forAllShrink(ImplicitlyUnwrappedOptional<A>.arbitrary, shrinker: ImplicitlyUnwrappedOptional<A>.shrink, f: { bl in
 			return pf(bl.map(wit))
 		})
 	}
 }
 
-extension LazyBidirectionalCollection where Base : protocol<CollectionType, Arbitrary>, Base.Index : BidirectionalIndexType {
-	public static var arbitrary : Gen<LazyBidirectionalCollection<Base>> {
-		return Base.arbitrary.fmap(lazy)
+private func lazy<S : CollectionType>(s: S) -> LazyCollection<S> {
+    return s.lazy
+}
+
+private func lazy<S : SequenceType>(s: S) -> LazySequence<S> {
```

Comment:
Maybe lets get rid of these [unless there's some overriding logic here?).  Just invoke 'em in their respective closures.